src/libponyrt/sched/scheduler.c

```diff
@@ -746,13 +777,29 @@ static pony_actor_t* steal(scheduler_t* sched)
  */
 static void run(scheduler_t* sched)
 {
+  last_cd_tsc = 0;
   pony_actor_t* actor = pop_global(sched);
   if (DTRACE_ENABLED(ACTOR_SCHEDULED) && actor != NULL) {
     DTRACE2(ACTOR_SCHEDULED, (uintptr_t)sched, (uintptr_t)actor);
   }
 
   while(true)
   {
+    // if we're scheduler 0 and cycle detection is enabled
+    if((sched->index == 0) && !ponyint_actor_getnoblock())
```

Comment:
does the value of `ponyint_actor_getnoblock` change during the course of a run? if not, these cases should probably be flipped.